corehq/apps/locations/templates/locations/manage/partials/drilldown_location_widget.html

```diff
@@ -15,6 +15,10 @@
             data: {'results': $(e).select2('data')},
         }
         $('#{{ css_id }}').select2(options);
+        option_ids = _.map(options.data.results, function(option) { return option.id })
+        if(!option_ids.includes($('#{{ css_id }}').val())) {
+            $('#{{ css_id }}').val(null);
+        }
```

Comment:
Sorry for duplication again, I think I fixed it in [here](https://github.com/dimagi/commcare-hq/pull/14163/files#diff-365eab9e49926ea6eed7f768d4705a27R38). If the value for primary location is not in the available options, then it is set as [first option](https://github.com/dimagi/commcare-hq/pull/14163/files#diff-365eab9e49926ea6eed7f768d4705a27R38) from available option. Otherwise, it is set to [null](https://github.com/dimagi/commcare-hq/pull/14163/files#diff-365eab9e49926ea6eed7f768d4705a27R32)